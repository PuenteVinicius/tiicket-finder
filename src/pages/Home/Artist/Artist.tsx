import { Box, Grid, Heading, Text, Button } from "grommet";
import IArtist from "../../../models/artist";
import { IAtractionClassifications } from "../../../models/atraction";

const Artist = (props: IArtist) => {
  let highlightImage: string = "";
  const { artist } = props;

  if (artist.images) {
    highlightImage = `url(${artist.images[0].url})`;
  }

  const showGenreSubgenre = (
    classifications: IAtractionClassifications[] | undefined
  ) => {
    if (!classifications) return;
    return classifications.map((classification: IAtractionClassifications) => (
      <Text
        alignSelf="start"
        margin="medium"
        color="white"
        weight="bold"
        size="xxlarge"
      >
        {classification.genre.name} / {classification.subGenre.name}
      </Text>
    ));
  };


  const showExeternalLinks = (
    externalLinks: any | undefined
  ) => {
    if (!externalLinks) return;
    return Object.keys(externalLinks).map((key: string, index: number) => 
      <Button
        key={key}
        margin="xsmall"
        href={externalLinks[key][0].url}
        primary
        label={key}
      />
    )};

  return (
    <Grid
      rows={["small", "small", "small", "small", "small", "small"]}
      columns={["small", "small", "small", "small", "small", "small"]}
      areas={[
        { name: "info", start: [0, 0], end: [3, 1] },
        { name: "highlight", start: [4, 0], end: [5, 1] },
      ]}
      gap="small"
    >
      <Box gridArea="highlight" background={highlightImage} />
      <Box gridArea="info" background="black">
        <Heading
          margin="medium"
          color="white"
          level="1"
          size="large"
          alignSelf="start"
        >
          {artist.name}
        </Heading>
        {showGenreSubgenre(artist.classifications)}
        <Text
          alignSelf="start"
          margin="medium"
          color="white"
          weight="bold"
          size="xxlarge"
        >
        {showExeternalLinks(artist.externalLinks)}
        </Text>
      </Box>
    </Grid>
  );
};

export default Artist;
