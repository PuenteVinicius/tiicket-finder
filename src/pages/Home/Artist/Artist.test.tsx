import { render } from "@testing-library/react";
import "jest-styled-components";
import TicketMasterApi from "../../../api/ticket-master/ticket-master";
import IAtraction from "../../../domain/atraction";
import Artist from "./Artist";

const ARTIST_NAME = "Dua Lipa";


describe("Artist component", () => {
  let artistComponent: any;

  beforeEach(() => {
    let atraction: IAtraction | string; 
    TicketMasterApi.getArtist(ARTIST_NAME).then(response => {
      atraction = response;
      artistComponent = render(<Artist artist={atraction} />);
    })
  });

  it("renders home correctly", () => {
    expect(artistComponent).toMatchSnapshot();
  });

});
