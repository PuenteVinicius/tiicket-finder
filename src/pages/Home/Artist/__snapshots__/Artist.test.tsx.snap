// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders home correctly 1`] = `undefined`;
